sphinxcontrib/drawio/__init__.py

```diff
@@ -209,6 +213,10 @@ def _drawio_export(self, input_abspath, options, out_filename):
             binary_path = builder.config.drawio_binary_path
         elif platform.system() == "Windows":
             binary_path = r"C:\Program Files\draw.io\draw.io.exe"
+        elif platform.system().startswith('CYGWIN_NT'):
+            binary_path = "/cygdrive/c/Program Files/draw.io/draw.io.exe"
+            export_abspath_cygwin = Path(re.sub('/cygdrive/(?P<letter>[a-z])/', '\g<letter>:/', self.imagedir, 1)) / sha_key / out_filename
```

Comment:
Setting the export abspath here overrides some of the earlier checks.

It would be preferable to add these _before_ L198, which also removes the issue with duplicating the `/ sha_key / out_filename`, and allows the "has this file been exported already" checks on L206-210 to be run on the cygwin path.